R/expectation.R

```diff
@@ -199,7 +190,13 @@ as.expectation.expectation <- function(x, srcref = NULL) {
 
 #' @export
 as.expectation.error <- function(x, srcref = NULL) {
-  expectation("error", cnd_message(x), srcref, trace = x[["trace"]])
+  if (is.null(x$call)) {
+    msg <- paste0("Error: ", cnd_message(x))
+  } else {
+    msg <- paste0("Error in ", expr_deparse(x$call[1]), ": ", cnd_message(x))
```

Comment:
`expr_deparse()` might return multiple lines in case of quoted `{` and inlined functions. In rlang complex expressions are skipped. I guess we want to strip arguments? The call should also be backticked.

Maybe it's easier to wait for rlang to export the relevant formatter?

Reply:
Shouldn’t the `[1]` be sufficient?

Reply:
Not always because the CAR might be a complex expression. You could use `as_label()` instead.

Reply:
Oooh yeah. Probably unlikely to affect us, but better to be safe.

Reply:
Did we want to also backtick the call for consistency with rlang errors?

Reply:
Do we consistently backtick code? I thought we only did that for argument names?

Reply:
rlang is following cli formatting:

```r
cli::cli_text("{.code foo()}")
#> `foo()`
```

Personally I like the consistency with markdown syntax.

Reply:
Makes sense. I'll fix momentarily.